components/match2/wikis/splitgate/match_group_input_custom.lua

```diff
@@ -36,388 +31,170 @@ local CustomMatchGroupInput = {}
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
 		end
+
+		table.insert(maps, map)
+		match[key] = nil
 	end
 
-	for index = 1, opponentNumber do
-		if not match['opponent' .. index].score and foundScores then
-			match['opponent' .. index].score = newScores[index] or 0
-		end
+	return maps
+end
+
+---@param bestofInput string|integer?
+---@return integer?
+function MatchFunctions.getBestOf(bestofInput)
+	local bestof = tonumber(bestofInput)
+
+	if bestof then
+		Variables.varDefine('bestof', bestof)
+		return bestof
 	end
 
-	return match
+	return tonumber(Variables.varDefault('bestof')) or DEFAULT_BESTOF
 end
 
 ---@param match table
 ---@return table
-function matchFunctions.getTournamentVars(match)
-	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'team'))
-	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+function MatchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode'), DEFAULT_MODE)
+	return MatchGroupInputUtil.getCommonTournamentVars(match)
+end
 
-	return MatchGroupInput.getCommonTournamentVars(match)
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
 end
 
 ---@param match table
 ---@return table
-function matchFunctions.getVodStuff(match)
-	match.stream = Streams.processStreams(match)
-	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
-
-	match.links = {}
-	local links = match.links
-	if match.esl then links.esl = 'https://play.eslgaming.com/match/' .. match.esl end
-	if match.stats then links.stats = match.stats end
-
-	-- apply vodgames
-	for index = 1, MAX_NUM_VODGAMES do
-		local vodgame = match['vodgame' .. index]
-		if not Logic.isEmpty(vodgame) then
-			local map = match['map' .. index] or {}
-			map.vod = map.vod or vodgame
-			match['map' .. index] = map
-		end
-	end
-	return match
+function MatchFunctions.getLinks(match)
+	return {
+		esl = match.esl and ('https://play.eslgaming.com/match/' .. match.esl) or nil,
+		stats = match.stats,
+	}
 end
 
 ---@param match table
 ---@return table
-function matchFunctions.getExtraData(match)
-	match.extradata = {
-		mapveto = matchFunctions.getMapVeto(match),
-		mvp = MatchGroupInput.readMvp(match),
+function MatchFunctions.getExtraData(match)
+	return {
+		mapveto = MatchFunctions.getMapVeto(match),
+		mvp = MatchGroupInputUtil.readMvp(match),
 		isconverted = 0
 	}
-	return match
 end
 
 -- Parse the mapVeto input
 ---@param match table
 ---@return table[]
-function matchFunctions.getMapVeto(match)
+function MatchFunctions.getMapVeto(match)
```

Comment:
This is unused (no cases of `vetoes` or `vetoesby`), so let's nuke this before it gets used